tests/testsuite/artifact_dep.rs

```diff
@@ -670,10 +685,24 @@ fn build_script_with_bin_artifact_and_lib_false() {
         "#,
         )
         .build();
+
+    #[allow(deprecated)]
     p.cargo("build -Z bindeps")
         .masquerade_as_nightly_cargo(&["bindeps"])
         .with_status(101)
         .with_stderr_does_not_contain("[..]sentinel[..]")
+        .with_stderr_data(str![[r#"
+[LOCKING] 2 packages to latest compatible versions
+[COMPILING] bar v0.5.0 ([ROOT]/foo/bar)
+[COMPILING] foo v0.0.0 ([ROOT]/foo)
+error[E0433]: failed to resolve: use of undeclared crate or module `bar`
+...
+
+For more information about this error, try `rustc --explain E0433`.
+[ERROR] could not compile `foo` (build script) due to 1 previous error
+...
+
+"#]])
         .run();
 }
```

Comment:
If I'm reading this right, we added new output coverage we didn't have before.  What was the motivation?

Reply:
(ditto for other sentinel cases)

Reply:
I believe I did this so we could see that the actual was while still keeping the `does_not_contain` separate. I went a head and removed it.